packages/core/src/InstanceSettings.ts

```diff
@@ -40,8 +40,15 @@ export class InstanceSettings {
 
 	readonly instanceId = this.generateInstanceId();
 
-	/** Always `leader` in single-main setup. `leader` or `follower` in multi-main setup. */
-	private instanceRole: InstanceRole = 'unset';
+	/**
+	 * A main is:
+	 * - `unset` during bootup,
+	 * - `leader` after bootup in single-main setup,
+	 * - `leader` or `follower` after bootup in multi-main setup.
+	 *
+	 * A non-main instance type (e.g. `worker`) is always `unset`.
```

Comment:
Should it instead be `worker` when the instance acts as a `worker`? Then it would be more "complete"

Reply:
So the role an instance has (unset, leader, follower) is only ever relevant for a main - and a role is separate from what kind of instance it is (main, worker, webhook, etc.) which is `instanceType`.

Ideally we should throw whenever attempting to access `instanceRole` (or its derived getters) from a non-main instance as this should never happen, but that'd be a bit outside the scope of this PR.